va/va.c

```diff
@@ -2128,3 +2128,41 @@ vaQueryVideoProcPipelineCaps(
     );
     return status;
 }
+
+VAStatus
+vaHWAsstCopy(
+    VADisplay         dpy,
+    VAHWCpObject      *dst,
+    VAHWCpObject      *src,
+    uint32_t          option,
+    uintptr_t*        cphandle
+)
+{
+  VAStatus va_status;
+  VADriverContextP ctx;
+  CHECK_DISPLAY(dpy);
+  ctx = CTX(dpy);
+
```

Comment:
Carl, I'd like to propose a more abstract solution, **not to stick with HW copy**.  For video<->system memory copy, the copy (with tiling/detiling) can can be done by SW or HW (EUs or blitter), right? If so, does it look reasonable for you to introduce vaAsstCopy(), the function will accept a flag which will define copy type (SW, EUs, Blitter)?

Reply:
I would better go with as simple as `vaCopy()`.

Reply:
Hi Carl, May I know what kinds of usage scenarios this vaCopy is for? We have buffer, image, surface definition.  And the input/output parameters are very limited, so what is the target usage for this vaCopy?

Reply:
hi Furong, vaCopy is used to faster some copy, use HW to handle the memory copy for better performance, for example, copy memory from local to system, local to local ...